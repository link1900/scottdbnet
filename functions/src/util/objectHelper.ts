import InternalServerError from '../error/InternalServerError';

export function unwrap<T>(value: T | undefined | null): T {
  if (value === null || value === undefined) {
    throw new InternalServerError('Attempted to unwrap value but was null or undefined');
  }
  return value;
}

export function exists<T>(value?: T): boolean {
  return !(value === null || value === undefined);
}
