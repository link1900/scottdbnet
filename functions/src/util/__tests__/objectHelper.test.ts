import { unwrap, exists } from '../objectHelper';

describe('objectHelper', () => {
  describe('#unwrap', () => {
    it(`unwraps maybe value something to value something`, () => {
      expect(unwrap('something')).toEqual('something');
    });

    it(`throw an exception when null`, () => {
      expect(() => unwrap(null)).toThrowErrorMatchingSnapshot();
    });

    it(`throw an exception when undefined`, () => {
      expect(() => unwrap(undefined)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('#exists', () => {
    const testCases = [
      { value: 'val1', expected: true },
      { value: { some: 'object' }, expected: true },
      { value: undefined, expected: false },
      { value: null, expected: false }
    ];

    testCases.forEach(testCase => {
      it(`exists find '${testCase.value}' to be ${testCase.expected}`, () => {
        expect(exists(testCase.value)).toEqual(testCase.expected);
      });
    });
  });
});
