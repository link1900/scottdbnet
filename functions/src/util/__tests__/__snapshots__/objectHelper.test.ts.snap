// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectHelper #unwrap throw an exception when null 1`] = `"Attempted to unwrap value but was null or undefined"`;

exports[`objectHelper #unwrap throw an exception when undefined 1`] = `"Attempted to unwrap value but was null or undefined"`;
